transformations/transformations/pool_allocator.py

```diff
@@ -630,7 +647,7 @@ def create_pool_allocator(self, routine, stack_size):
                 assign_pos = -1
 
             # Check for existing pointer assignment
-            if any(a.lhs == stack_ptr.name for a in assignments):  # pylint: disable=no-member
+            if any(a.lhs == f'{self.stack_argument_name}_{self.stack_ptr_name}' for a in assignments):  # pylint: disable=no-member
```

Comment:
The pylint annotation seems no longer necessary, here. However, test coverage suggests that this never evaluates to true albeit it did before your changes. Since the tests include a case where pointer assignments exist already, I'm suspecting something might not be working as expected here. Would you mind double-checking the generated driver code for the various test parameterisations to make sure the stack allocation and assignment looks the way it's supposed to look?